### PR TITLE
Changed the server greeting buffer from 256 to 1024.

### PR DESCRIPTION
**Pull request by delebota**
Allows for more information to be shared with users in multiplayer. As discussed in #4193

### CODE REVIEW

**Comment by janisozaur** [comment]
@delebota please see https://github.com/OpenRCT2/OpenRCT2/wiki/Commit-Messages about commit message length.

File: src/network/network.cpp [hunk 1]
```diff
@@ -2152,13 +2152,13 @@ void network_chat_show_server_greeting()
...
 	if (!str_is_null_or_empty(greeting)) {
-		static char greeting_formatted[256];
+		static char greeting_formatted[1024];
 		char* lineCh = greeting_formatted;
 		greeting_formatted[0] = 0;
 		lineCh = utf8_write_codepoint(lineCh, FORMAT_OUTLINE);
 		lineCh = utf8_write_codepoint(lineCh, FORMAT_GREEN);
 		char* ptrtext = lineCh;
-		safe_strcpy(lineCh, greeting, 240);
+		safe_strcpy(lineCh, greeting, 1000);
```
Comment (janisozaur) [inline src/network/network.cpp]: please include a short comment on why is it limited to below the buffer size.

File: src/network/network.cpp [hunk 2]
```diff
@@ -2152,13 +2152,13 @@ void network_chat_show_server_greeting()
...
 	if (!str_is_null_or_empty(greeting)) {
-		static char greeting_formatted[256];
+		static char greeting_formatted[1024];
 		char* lineCh = greeting_formatted;
 		greeting_formatted[0] = 0;
 		lineCh = utf8_write_codepoint(lineCh, FORMAT_OUTLINE);
 		lineCh = utf8_write_codepoint(lineCh, FORMAT_GREEN);
 		char* ptrtext = lineCh;
-		safe_strcpy(lineCh, greeting, 240);
+		safe_strcpy(lineCh, greeting, 1000); // Limit to 1000 characters so we don't overflow the buffer
```
Comment (IntelOrca) [inline src/network/network.cpp]: Use `CHAT_INPUT_SIZE - 24` ?

File: src/network/network.cpp [hunk 3]
```diff
@@ -2152,13 +2152,13 @@ void network_chat_show_server_greeting()
 {
 	const char* greeting = gConfigNetwork.server_greeting;
 	if (!str_is_null_or_empty(greeting)) {
-		static char greeting_formatted[256];
+		static char greeting_formatted[1024];
```
Comment (IntelOrca) [inline src/network/network.cpp]: Use `CHAT_INPUT_SIZE ` ?

**Comment by delebota** [comment]
That was my original thought, but I thought there was issues in C when two files include each other. Seems like they already do though, so I will switch it now.

**Comment by IntelOrca** [comment]
It needs to be defined in a header file.

**Pull request merged**